fineract-progressive-loan/src/main/java/org/apache/fineract/portfolio/loanaccount/service/BuyDownFeeWritePlatformServiceImpl.java

```diff
@@ -210,6 +213,40 @@ public CommandProcessingResult buyDownFeeAdjustment(final Long loanId, final Lon
                 .withClientId(loan.getClientId()).withLoanId(loan.getId()).build();
     }
 
+    @Override
+    @Transactional
+    public Optional<LoanTransaction> reverseBuyDownFee(LoanTransaction buyDownFeeTransaction) {
+        LoanTransaction amortizationTransaction = null;
+        Loan loan = buyDownFeeTransaction.getLoan();
+        BigDecimal totalAmortizationAmount = BigDecimal.ZERO;
+
+        final List<LoanTransactionRelation> loanTransactionRelations = loanTransactionRelationRepository
```

Comment:
We dont need to fetch all the relations and go through on loan transactions (and effectively fetch them) to make this decision. Please write a simply query which returns the required information:
- boolean
- related transaction id